src/VASSAL/build/module/Chatter.java

```diff
@@ -21,92 +21,109 @@
-       JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
-  protected static final String MY_CHAT_COLOR = "myChatColor"; //$NON-NLS-1$
-  protected static final String OTHER_CHAT_COLOR = "otherChatColor"; //$NON-NLS-1$
-  protected static final String GAME_MSG_COLOR = "gameMessageColor"; //$NON-NLS-1$
-  protected static final String SYS_MSG_COLOR = "systemMessageColor"; //$NON-NLS-1$
-
-  protected Color gameMsg;
-  protected Color systemMsg;
-  protected Color myChat;
-  protected Color otherChat;
+  protected JScrollPane scroll = new ScrollPane(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+  protected JScrollPane scroll2 = new ScrollPane(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+  protected static final String MY_CHAT_COLOR    = "HTMLChatColor";          //$NON-NLS-1$ //BR// Different tags to "restart" w/ new default scheme
+  protected static final String OTHER_CHAT_COLOR = "HTMLotherChatColor";     //$NON-NLS-1$
+  protected static final String GAME_MSG1_COLOR  = "HTMLgameMessage1Color";  //$NON-NLS-1$
+  protected static final String GAME_MSG2_COLOR  = "HTMLgameMessage2Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG3_COLOR  = "HTMLgameMessage3Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG4_COLOR  = "HTMLgameMessage4Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG5_COLOR  = "HTMLgameMessage5Color";  //$NON-NLS-1$ //BR//
+  protected static final String SYS_MSG_COLOR    = "HTMLsystemMessageColor"; //$NON-NLS-1$
+  protected static final String HTML_COMPATIBILITY = "HTMLcompatibility";    //$NON-NLS-1$
+
+  protected Font myFont;
+
+  protected Color gameMsg, gameMsg2, gameMsg3, gameMsg4, gameMsg5;
+  protected Color systemMsg, myChat, otherChat;
+
+  protected boolean htmlCompatibility; //BR// Disable HTML parsing unless opted in, to maintain backwards compatibility
 
   public static String getAnonymousUserName() {
     return Resources.getString("Chat.anonymous"); //$NON-NLS-1$
   }
 
-  public Chatter() {
+  public Chatter() {    
     setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
-    conversation = new JTextArea(10, 80);
-    for (int i = 0; i < 10; ++i) {
-      conversation.append("\n"); //$NON-NLS-1$
+
+    //BR// Conversation is now a JTextPane w/ HTMLEditorKit to process HTML, which gives us HTML support "for free".
+    conversation = new JTextPane();
+    conversation.setPreferredSize(new Dimension(500, 120));
+    conversation.setContentType("text/html");
+    doc = (HTMLDocument) conversation.getDocument();
+    kit = (HTMLEditorKit) conversation.getEditorKit();
+
+    style = kit.getStyleSheet();
+    myFont = new Font("SansSerif", Font.PLAIN, 12); //BR// Will be overridden by the font from Chat preferences
+
+    for (int i = 0; i < 15; ++i) {
+      try {
+        kit.insertHTML(doc, doc.getLength(), "<br>", 0, 0, null);
+      } catch (BadLocationException ble) {
```

Comment:
Vassal's default code style is, else, catch, finally etc on a new line, i.e.

```
if (cond) {
  // foo
}
else {
}

try {
  // try
}
catch (Exception e) {
}
finally {
}

Reply:
fixed for next push